design/vela-core/rollout-design.md

```diff
@@ -165,26 +164,38 @@ upon. For example, the applicationDeployment controller will get the component f
  With that said, two controllers operate differently to extract the real workload. Here are the
   high level descriptions of how each works. 
  
-#### ApplicationDeployment takes control of the application during rolling out phase
-When an appDeployment is used to do application level rollout, **_one should apply the
- appDeployment CR before the application itself_**. This is to try to make sure that the
-  appDeployment controll has full control of the new application from the beginning.
- However, there is no kubernetes systematic way of acquiring a distributed lock for a controller.  
-- The appDeployment controller will make sure it marks itself as the owner of the
- application and changes the appDeployment CR state to be initialized. The application controller
-  can be aware of appDeployment and have build-in logics in case.
-- The appDeployment controller can change the application fields. For example, it can remove all
- the conflict traits, such as HPA, during the upgrade period. 
-- Upon a successful rollout, the appDeployment controller can remove all the traits in the old
-  application and leave no pods in the workload.
-- Upon a failed rollout, the condition is not determined, it could result in an unstable state.
+#### ApplicationDeployment workflow 
+When an appDeployment is used to do application level rollout, **the target application
+is not reconciled by the application controller yet**. This is to make sure  the
+appDeployment controller has the full control of the new application from the beginning.
+We will use a pre-defined annotation "app.oam.dev/rollout" that equals to "true" to facilitate
```

Comment:
```suggestion
We will use a pre-defined annotation "app.oam.dev/template" that equals to "true" to facilitate
```

I would suggest use some words like `template`/`dummy` or something that can obviously tell the CR won't really work.

Reply:
How about "rollout-template" cause I feel like "template" alone is too general

Reply:
`rollout-template` is fine